src/Microsoft.Tye.Hosting/ReplicaRegistry.cs

```diff
@@ -0,0 +1,126 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Microsoft.Tye.Hosting.Model;
+
+namespace Microsoft.Tye.Hosting
+{
+    public class ReplicaRegistry : IDisposable
+    {
+        private readonly IDictionary<ServiceType, IReplicaInstantiator> _replicaInstantiators;
+        private StateStore _store;
+
+        public ReplicaRegistry(Model.Application application, IDictionary<ServiceType, IReplicaInstantiator> replicaInstantiators)
+        {
+            _replicaInstantiators = replicaInstantiators;
+            _store = new StateStore(application);
+        }
+
+        public async Task Reset()
+        {
+            await PurgeAll();
+            _store.Reset(true, true);
```

Comment:
nitpick: these cases would be much more comprehensible/readable with named parmaters. `Reset(delete: true, create: false)` is much more appealing 👍